test/unit/index.js

```diff
@@ -0,0 +1,145 @@
+'use strict'
+
+const {test, threw} = require('tap')
+
+const config = require('../../lib/config.js')
+const {handler} = require('../../index.js')
+const {s3} = require('../../lib/transformer.js')
+const {setProperty} = require('../../lib/utils.js')
+
+const BUCKET_NAME = 'random_name'
+const ERROR_MESSAGE = 'A connection-based error occurred that will not be retried.'
+  + ' See meta data for details.'
+const ERROR_STATUS_CODE = 403
+const FILE_NAME = 'log+File.json'
+const LOG_LINE = 'test log'
+const TIMESTAMP = Date.now()
+const EVENT_DATA = {
+  Records: [{
+    eventTime: new Date(TIMESTAMP)
+  , s3: {
+      bucket: {
+        name: BUCKET_NAME
+      }
+    , object: {
+        key: FILE_NAME
+      }
+    }
+  }]
+}
+
+// NOTE: We will NOT test the use of ENV vars directly here since they are cumbersome
+// to work with (they can never be updated). For this, we will trust that env-config
+// will do its job and read from the envrionment properly, but we will set the
+// config's keys manually in these tests.
+config.validateEnvVars()
+config.set('ingestion-key', 'required-for-anything-to-work')
+
+test('handler is the main method in lambda functions', async (t) => {
```

Comment:
these tests don't appear to be testing what they say, other than that the logger is invoked. we're missing tests that validate the requests to the ingestion endpoint (you can use `nock` for this). It doesn't have to be too exhaustive but would be good to validate that the proper data is included in the logger request.

Reply:
👍  yeah the `@logdna/logger` should be a good guide to follow, again just need some basic tests that intercept the request and validate that the right data is being sent to our endpoint. This test file can be moved into `test/integration/index.js`